app/queries/drug_stocks_query.rb

```diff
@@ -0,0 +1,103 @@
+class DrugStocksQuery
+  CACHE_VERSION = 1
+
+  def initialize(facilities:, for_end_of_month:)
+    @facilities = facilities
+    @for_end_of_month = for_end_of_month
+    # assuming that all facilities on the page have the same protocol
+    @protocol = @facilities.first.protocol
+    @state = @facilities.first.state
+  end
+
+  def protocol_drugs_by_category
+    @protocol_drugs_by_category ||= @protocol.protocol_drugs
+      .where(stock_tracked: true)
+      .order(:name, :dosage)
```

Comment:
Why order if we're sorting right under?

Reply:
Oh good catch :) Missed this while refactoring.

Reply:
Fixed in f31505e40e228389038f3b069073479562337d12.